File: backend/internal/service/system/admin/v1/base_redact_storage_policy.go
package admin

import (
	"context"
	"fmt"

	adminv1 "github.com/liujitcn/kratos-admin/backend/api/gen/go/system/admin/v1"
	biz "github.com/liujitcn/kratos-admin/backend/internal/biz/system/admin"
	"github.com/liujitcn/kratos-core/errorsx"

	"github.com/go-kratos/kratos/v3/log"
	"google.golang.org/protobuf/types/known/emptypb"
)

// BaseRedactStoragePolicyService 提供入库脱敏策略管理接口。
type BaseRedactStoragePolicyService struct {
	adminv1.UnimplementedBaseRedactStoragePolicyServiceServer
	baseRedactStoragePolicyCase *biz.BaseRedactStoragePolicyCase
}

// NewBaseRedactStoragePolicyService 创建入库脱敏策略服务。
func NewBaseRedactStoragePolicyService(baseRedactStoragePolicyCase *biz.BaseRedactStoragePolicyCase) *BaseRedactStoragePolicyService {
	return &BaseRedactStoragePolicyService{baseRedactStoragePolicyCase: baseRedactStoragePolicyCase}
}

// PageBaseRedactStoragePolicy 分页查询入库脱敏策略。
func (s *BaseRedactStoragePolicyService) PageBaseRedactStoragePolicy(ctx context.Context, req *adminv1.PageBaseRedactStoragePolicyRequest) (*adminv1.PageBaseRedactStoragePolicyResponse, error) {
	result, err := s.baseRedactStoragePolicyCase.PageBaseRedactStoragePolicy(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("PageBaseRedactStoragePolicy %v", err))
		return nil, errorsx.WrapInternal(err, "分页查询入库脱敏策略失败")
	}
	return result, nil
}

// GetBaseRedactStoragePolicy 查询入库脱敏策略详情。
func (s *BaseRedactStoragePolicyService) GetBaseRedactStoragePolicy(ctx context.Context, req *adminv1.GetBaseRedactStoragePolicyRequest) (*adminv1.BaseRedactStoragePolicyForm, error) {
	result, err := s.baseRedactStoragePolicyCase.GetBaseRedactStoragePolicy(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("GetBaseRedactStoragePolicy %v", err))
		return nil, errorsx.WrapInternal(err, "查询入库脱敏策略详情失败")
	}
	return result, nil
}

// CreateBaseRedactStoragePolicy 批量创建入库脱敏策略。
func (s *BaseRedactStoragePolicyService) CreateBaseRedactStoragePolicy(ctx context.Context, req *adminv1.CreateBaseRedactStoragePolicyRequest) (*emptypb.Empty, error) {
	err := s.baseRedactStoragePolicyCase.CreateBaseRedactStoragePolicy(ctx, req.GetBaseRedactStoragePolicy())
	if err != nil {
		log.Error(fmt.Sprintf("CreateBaseRedactStoragePolicy %v", err))
		return nil, errorsx.WrapInternal(err, "创建入库脱敏策略失败")
	}
	return new(emptypb.Empty), nil
}

// UpdateBaseRedactStoragePolicy 批量更新入库脱敏策略。
func (s *BaseRedactStoragePolicyService) UpdateBaseRedactStoragePolicy(ctx context.Context, req *adminv1.UpdateBaseRedactStoragePolicyRequest) (*emptypb.Empty, error) {
	err := s.baseRedactStoragePolicyCase.UpdateBaseRedactStoragePolicy(ctx, req.GetBaseRedactStoragePolicy())
	if err != nil {
		log.Error(fmt.Sprintf("UpdateBaseRedactStoragePolicy %v", err))
		return nil, errorsx.WrapInternal(err, "更新入库脱敏策略失败")
	}
	return new(emptypb.Empty), nil
}

// DeleteBaseRedactStoragePolicy 删除入库脱敏策略。
func (s *BaseRedactStoragePolicyService) DeleteBaseRedactStoragePolicy(ctx context.Context, req *adminv1.DeleteBaseRedactStoragePolicyRequest) (*emptypb.Empty, error) {
	err := s.baseRedactStoragePolicyCase.DeleteBaseRedactStoragePolicy(ctx, req.GetId())
	if err != nil {
		log.Error(fmt.Sprintf("DeleteBaseRedactStoragePolicy %v", err))
		return nil, errorsx.WrapInternal(err, "删除入库脱敏策略失败")
	}
	return new(emptypb.Empty), nil
}

// SetBaseRedactStoragePolicyStatus 设置入库脱敏策略状态。
func (s *BaseRedactStoragePolicyService) SetBaseRedactStoragePolicyStatus(ctx context.Context, req *adminv1.SetBaseRedactStoragePolicyStatusRequest) (*emptypb.Empty, error) {
	err := s.baseRedactStoragePolicyCase.SetBaseRedactStoragePolicyStatus(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("SetBaseRedactStoragePolicyStatus %v", err))
		return nil, errorsx.WrapInternal(err, "设置入库脱敏策略状态失败")
	}
	return new(emptypb.Empty), nil
}

// ListBaseRedactStorageTable 查询包含租户ID字段的数据表列表。
func (s *BaseRedactStoragePolicyService) ListBaseRedactStorageTable(ctx context.Context, req *adminv1.ListBaseRedactStorageTableRequest) (*adminv1.ListBaseRedactStorageTableResponse, error) {
	result, err := s.baseRedactStoragePolicyCase.ListBaseRedactStorageTable(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("ListBaseRedactStorageTable %v", err))
		return nil, errorsx.WrapInternal(err, "查询可入库脱敏数据表失败")
	}
	return result, nil
}

// ListBaseRedactStorageColumn 查询可入库脱敏的字符串字段列表。
func (s *BaseRedactStoragePolicyService) ListBaseRedactStorageColumn(ctx context.Context, req *adminv1.ListBaseRedactStorageColumnRequest) (*adminv1.ListBaseRedactStorageColumnResponse, error) {
	result, err := s.baseRedactStoragePolicyCase.ListBaseRedactStorageColumn(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("ListBaseRedactStorageColumn %v", err))
		return nil, errorsx.WrapInternal(err, "查询可入库脱敏字段失败")
	}
	return result, nil
}

File: frontend/admin/packages/modules/system/src/api/system/admin/v1/base_redact_storage_policy.ts
import service from "@liujitcn/kratos-admin-core/request";
import type {
  BaseRedactStoragePolicyForm,
  BaseRedactStoragePolicyService,
  CreateBaseRedactStoragePolicyRequest,
  DeleteBaseRedactStoragePolicyRequest,
  GetBaseRedactStoragePolicyRequest,
  ListBaseRedactStorageColumnRequest,
  ListBaseRedactStorageColumnResponse,
  ListBaseRedactStorageTableRequest,
  ListBaseRedactStorageTableResponse,
  PageBaseRedactStoragePolicyRequest,
  PageBaseRedactStoragePolicyResponse,
  SetBaseRedactStoragePolicyStatusRequest,
  UpdateBaseRedactStoragePolicyRequest
} from "@liujitcn/kratos-admin-system/rpc/system/admin/v1/base_redact_storage_policy";
import type { Empty } from "@liujitcn/kratos-admin-system/rpc/google/protobuf/empty";

const BASE_REDACT_STORAGE_POLICY_URL = "/v1/admin/base/redact-storage-policy";

/** 管理端入库脱敏策略服务。 */
export class BaseRedactStoragePolicyServiceImpl implements BaseRedactStoragePolicyService {
  /** 查询入库脱敏策略分页列表。 */
  PageBaseRedactStoragePolicy(request: PageBaseRedactStoragePolicyRequest): Promise<PageBaseRedactStoragePolicyResponse> {
    return service<PageBaseRedactStoragePolicyRequest, PageBaseRedactStoragePolicyResponse>({ url: BASE_REDACT_STORAGE_POLICY_URL, method: "get", params: request });
  }

  /** 查询入库脱敏策略详情。 */
  GetBaseRedactStoragePolicy(request: GetBaseRedactStoragePolicyRequest): Promise<BaseRedactStoragePolicyForm> {
    return service<GetBaseRedactStoragePolicyRequest, BaseRedactStoragePolicyForm>({ url: `${BASE_REDACT_STORAGE_POLICY_URL}/${request.id}`, method: "get" });
  }

  /** 批量创建入库脱敏策略。 */
  CreateBaseRedactStoragePolicy(request: CreateBaseRedactStoragePolicyRequest): Promise<Empty> {
    return service<CreateBaseRedactStoragePolicyRequest, Empty>({ url: BASE_REDACT_STORAGE_POLICY_URL, method: "post", data: request });
  }

  /** 批量更新入库脱敏策略。 */
  UpdateBaseRedactStoragePolicy(request: UpdateBaseRedactStoragePolicyRequest): Promise<Empty> {
    return service<UpdateBaseRedactStoragePolicyRequest, Empty>({ url: BASE_REDACT_STORAGE_POLICY_URL, method: "put", data: request });
  }

  /** 删除入库脱敏策略。 */
  DeleteBaseRedactStoragePolicy(request: DeleteBaseRedactStoragePolicyRequest): Promise<Empty> {
    return service<DeleteBaseRedactStoragePolicyRequest, Empty>({ url: `${BASE_REDACT_STORAGE_POLICY_URL}/${request.id}`, method: "delete" });
  }

  /** 设置入库脱敏策略状态。 */
  SetBaseRedactStoragePolicyStatus(request: SetBaseRedactStoragePolicyStatusRequest): Promise<Empty> {
    return service<SetBaseRedactStoragePolicyStatusRequest, Empty>({ url: `${BASE_REDACT_STORAGE_POLICY_URL}/${request.id}/status`, method: "put", data: request });
  }

  /** 查询包含租户ID字段的数据表列表。 */
  ListBaseRedactStorageTable(request: ListBaseRedactStorageTableRequest): Promise<ListBaseRedactStorageTableResponse> {
    return service<ListBaseRedactStorageTableRequest, ListBaseRedactStorageTableResponse>({ url: `${BASE_REDACT_STORAGE_POLICY_URL}/tables`, method: "get", params: request });
  }

  /** 查询可入库脱敏的字符串字段列表。 */
  ListBaseRedactStorageColumn(request: ListBaseRedactStorageColumnRequest): Promise<ListBaseRedactStorageColumnResponse> {
    return service<ListBaseRedactStorageColumnRequest, ListBaseRedactStorageColumnResponse>({ url: `${BASE_REDACT_STORAGE_POLICY_URL}/columns`, method: "get", params: request });
  }
}

/** defBaseRedactStoragePolicyService 入库脱敏策略服务实例。 */
export const defBaseRedactStoragePolicyService = new BaseRedactStoragePolicyServiceImpl();
